### ISSUE TITLE
[BUG] new RKE2 releases do not show CSI Options in the UI for vsphere cluster

### ISSUE THREAD

**Issue by Zappelphilipp** [rancher/rancher#44754]
**Rancher Server Setup**
- Rancher version: tested on 2.8.2 & 2.7.10
- Installation option (Docker install/Helm Chart): Helm tested on RKE2 1.27.11 & 1.26.14

- tested for Downstreamclusters 1.27.11 & 1.26.14 vsphere based clusters
-->

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
Admin

**Describe the bug**
During setup of new vsphere clusters under `Add-on Config` there is no field "vSphere CSI Configuration" anymore as well the following error in the browser development console occurs:

`Failed to fetch or process chart info for rancher-vsphere-csi`
![grafik](https://github.com/rancher/rancher/assets/11159698/ee65cfb5-6af9-4dab-8bf8-202eb8b0f130)

also after the clusterdeployment there are some vsphere-csi-controller pods in CrashLoopBackOff state. 
**To Reproduce**

1. Create new vsphere based downstream cluster 
2. At `Cluster Configuration` select `Basics`
3. select Kubernetes Version 1.27.11 or 1.26.14
4. select vsphere as Cloud Provider
5. switch to `Add-on Config` 
6. expect vSphere CPI Configuration and vSphere CSI Configuration but CSI Configuration form is missing:

![grafik](https://github.com/rancher/rancher/assets/11159698/8d95bfd4-eb18-4a7e-ba3b-f06f7c202081)

**Result**

**Expected Result**
can be tested by following the same procedure but selecting 1.27.10 or below or 1.26.14 or below. 

expected is the configuration dialog of  vSphere CPI Configuration and vSphere CSI Configuration as well as not getting errors in the browser console. 

![grafik](https://github.com/rancher/rancher/assets/11159698/085f185c-a2c3-4fc0-bffd-a9a87cb971a9)

**Additional context**
I know there is some documentation about the new out-of-tree implementation here: https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/migrate-to-an-out-of-tree-cloud-provider/migrate-to-out-of-tree-vsphere

but it's quite confusing to me so i am not sure if the new behavior is as designed as well as the documentation states that there should be the CPI as well as CSI configured during cluster creation AND the CPI and CSI plugin should be installed afterwards as App in the Application menu.

1. why both? what exactly is the difference between the `rancher-vsphere-csi` /cpi which is configured during installation VS the `vsphere-csi` /cpi which can be installed afterwards in the application section?
2. is one of the procedures using the in-tree implementation? for my understanding both of them are the out-of-tree implementation from vmware.

EDIT: tried installing the CSI App after the deployment but it fails:

```
Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: ServiceAccount "vsphere-csi-controller" in namespace "kube-system" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "vsphere-csi": current value is "rancher-vsphere-csi"
```

**Comment by tariksimsek** [comment]
Any news about this issue ? thanks

**Comment by chrpinedo** [comment]
I am facing this issue too and the CSI shows errors

**Comment by Zappelphilipp** [comment]
for everybody facing the same issue: The only workaround i found is to use the older versions which do not have the problem but are marked as `deprecated`

using 1.26.14 is also affected so even tough it's not deprecated, using it would not solve the issue.
![grafik](https://github.com/rancher/rancher/assets/11159698/842af827-1348-4204-9e39-12b7b437665e)

**Comment by gaktive** [comment]
As UI understands, this ties into getting https://github.com/rancher/rke2/issues/5217 in and there may be overlapped testing in https://github.com/rancher/rancher/issues/43980 so Dashboard doesn't need to update anything. Do keep me posted if this changes and we should get a ticket in rancher/dashboard

**Comment by tariksimsek** [comment]
How to solve this issue ? I upgrade rke2 to 1.27 and configs are messed up. How can they make a mistake about this i don't understand.  This is a major issue

**Comment by rancher-max** [comment]
This has been validated by the k3s/rke2 team as fixed with the latest KDM changes. Therefore, this will be fixed in live environments for March k8s patches of rke2.

**Comment by snasovich** [comment]
Closing per the above comment as March RKE/K3s versions are included in upcoming 2.8.3 release.
